File: 백준/Bronze/2484. 주사위 네개/주사위 네개.js
const fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input.shift());
let answer = 0;

for (let i=0; i<n; i++) {
  let newArr = input[i].trim().split(' ').map(Number);
  let newMap = new Map();
  let total = 0;

  for (num of newArr) {
    if (!newMap.has(num)) {
      newMap.set(num,1);
    } else {
      newMap.set(num,newMap.get(num)+1);
    }
  }

  newMapArr = [...newMap];
  newMapArr.sort((a,b) => b[1]-a[1]);
  let k_arr = [...newMap.keys()];
  let v_arr = [...newMap.values()];

  for (info of newMapArr) {
    let [k,v] = [info[0],info[1]]
    
    if (v === 4) {
      total = 50000 + k * 5000;
    } 
    else if (v === 3) {
      total = 10000 + k*1000;
      break
    }
    else if (v === 2) {
      let t_cnt = v_arr.filter(elem => elem === 2).length;

      if (t_cnt === 2) {
        total = 2000 + k_arr[0]*500 + k_arr[1]*500;
        break
      } else {
        total = 1000 + k*100;
        break
      }
    } else {
      total = Math.max(...k_arr) * 100
    }
  }
  answer = Math.max(answer,total); 
}
console.log(answer);

File: 백준/Bronze/2484. 주사위 네개/주사위 네개.py
import sys
input = sys.stdin.readline

n = int(input().rstrip())
answer = 0
for _ in range(n):
    dice = {}
    row = list(map(int,input().split()))
    for r in row:
        if r not in dice:
            dice[r] = 1
        else:
            dice[r] += 1
    
    dice_arr = list(dice.items())
    dice_arr.sort(key=lambda x:x[1], reverse=True)
    
    for key,value in dice_arr:
        if value == 4:
            total = 50000 + key*5000
        elif value == 3:
            total = 10000 + key*1000
            break
        elif value == 2:
            if list(dice.values()).count(2) == 2:
                tmp = list(dice.keys())
                total = 2000 + tmp[0] * 500 + tmp[1] * 500
                break
            else:
                total = 1000 + key * 100
        elif list(dice.values()).count(1) == 4:
            tmp = max(list(dice.keys()))
            total = tmp * 100

    answer = max(answer,total)

print(answer)